src/Items/ItemBoat.h

```diff
@@ -98,6 +98,12 @@ class cItemBoatHandler :
 		cBoat * Boat = new cBoat(x + 0.5, y + 0.5, z + 0.5);
 		Boat->Initialize(*a_World);
 
+		// Remove boat from players hand
+		if (!a_Player->IsGameModeCreative())
```

Comment:
You can't place blocks in spectator mode; you have no hotbar.  There [already seems to be a check](https://github.com/cuberite/cuberite/blob/01d9beee9ee60aa1ead52f7796c762a8372b9ca4/src/ClientHandle.cpp#L1482) that prevents using items in spectator mode.  Even if that check isn't complete, the check shouldn't be put into boats specifically; it should be in a more general location (as it already seems to be).